feature-libs/cart/main/core/services/multi-cart-state-persistence.service.ts

```diff
@@ -60,6 +69,19 @@ export class MultiCartStatePersistenceService implements OnDestroy {
     } else {
       this.store.dispatch(new CartActions.SetActiveCartId(''));
     }
+    this.dispatchActiveCartBrowserStorageChangeEvent(state);
+  }
+
+  protected onPersist(state: { active: string } | undefined) {
```

Comment:
When `onPersist` is called and event is created, the cart-lib has been already loaded. So, we don't care about whether `ActiveCartBrowserStorageChange`, we care about whether the cart-lib loaded or not. I think the logic is this: if cart-lib is load, we just use the stream from `ActiveCartFacade.getActive()`. If the cart-lib is not loaded, we check the `getCartStateFromBrowserStorage`, and user login or not, then decide which stream to use.